pkg/controller/operands/kubevirt_test.go

```diff
@@ -1569,15 +1582,107 @@ var _ = Describe("KubeVirt Operand", func() {
 				WithHostPassthroughCPU: &disabled,
 				WithHostModelCPU:       &enabled,
+				HypervStrictCheck:      &disabled,
+				SRIOVLiveMigration:     &enabled,
 			}
 			fgList := getKvFeatureGateList(fgs)
-			Expect(fgList).To(HaveLen(2))
-			Expect(fgList).Should(ContainElements(HotplugVolumesGate, kvWithHostModelCPU))
-			Expect(fgList).ShouldNot(ContainElements(kvWithHostPassthroughCPU))
+			Expect(fgList).To(HaveLen(3))
+			Expect(fgList).Should(ContainElements(HotplugVolumesGate, kvWithHostModelCPU, SRIOVLiveMigrationGate))
+			Expect(fgList).ShouldNot(ContainElements(kvWithHostPassthroughCPU, kvHypervStrictCheck))
+		})
+	})
+
+	Context("filterOutDisabledFeatureGates", func() {
+		It("should remove all gates if none are enabled on the request", func() {
+			disabled := false
+			fgs := &hcov1beta1.HyperConvergedFeatureGates{
+				HotplugVolumes:         &disabled,
+				WithHostPassthroughCPU: &disabled,
+				WithHostModelCPU:       &disabled,
+				HypervStrictCheck:      &disabled,
+				SRIOVLiveMigration:     &disabled,
+			}
+
+			changed, result := filterOutDisabledFeatureGates(requestWithFeatureGates(fgs), []string{
+				HotplugVolumesGate,
+				kvWithHostPassthroughCPU,
+				kvWithHostModelCPU,
+				kvHypervStrictCheck,
+				SRIOVLiveMigrationGate,
+			}, make([]string, 0))
+			Expect(changed).To(BeTrue())
+			Expect(result).To(BeEmpty())
+		})
+		It("should remove only gates disabled on the request", func() {
+			enabled := true
+			disabled := false
+			fgs := &hcov1beta1.HyperConvergedFeatureGates{
+				HotplugVolumes:   &disabled,
+				WithHostModelCPU: &enabled,
+			}
+
+			changed, result := filterOutDisabledFeatureGates(
+				requestWithFeatureGates(fgs), []string{
+					HotplugVolumesGate,
+					kvWithHostModelCPU,
+				}, make([]string, 0, 1))
+			Expect(changed).To(BeTrue())
+			Expect(result).To(ContainElement(kvWithHostModelCPU))
+		})
+		It("should remove all gates if request gates are nil", func() {
+			changed, result := filterOutDisabledFeatureGates(
+				requestWithFeatureGates(nil), []string{
+					HotplugVolumesGate,
+					kvWithHostModelCPU,
+				}, make([]string, 0))
+			Expect(changed).To(BeTrue())
+			Expect(result).To(BeEmpty())
+		})
+		It("should not remove gates if none are disabled", func() {
+			enabled := true
+			fgs := &hcov1beta1.HyperConvergedFeatureGates{
+				HotplugVolumes:         &enabled,
+				WithHostPassthroughCPU: &enabled,
+				WithHostModelCPU:       &enabled,
+				HypervStrictCheck:      &enabled,
+				SRIOVLiveMigration:     &enabled,
+			}
+
+			allGates := []string{HotplugVolumesGate,
+				kvWithHostPassthroughCPU,
+				kvWithHostModelCPU,
+				kvHypervStrictCheck,
+				SRIOVLiveMigrationGate,
+			}
+
+			changed, result := filterOutDisabledFeatureGates(
+				requestWithFeatureGates(fgs),
+				allGates,
+				make([]string, 0, len(allGates)))
+			Expect(changed).To(BeFalse())
+			Expect(result).To(Equal(allGates))
+		})
+		It("should remove gates that don't have a matching field", func() {
```

Comment:
@nunnatsa I'm not sure if that assumption is correct.
If my brain is not playing tricks on me the previous code did not remove gates that don't exist while it should have imo.
What do you think?

Reply:
I removed this test for now as it collided with existing behavior.